submission/submission-webapp/src/itest/kotlin/ac/uk/ebi/biostd/itest/test/submission/submit/SubmissionStorageModeTest.kt

```diff
@@ -0,0 +1,126 @@
+package ac.uk.ebi.biostd.itest.test.submission.submit
+
+import ac.uk.ebi.biostd.client.integration.commons.SubmissionFormat.TSV
+import ac.uk.ebi.biostd.client.integration.web.BioWebClient
+import ac.uk.ebi.biostd.common.config.PersistenceConfig
+import ac.uk.ebi.biostd.itest.common.SecurityTestService
+import ac.uk.ebi.biostd.itest.entities.SuperUser
+import ac.uk.ebi.biostd.itest.itest.ITestListener.Companion.fireSubmissionPath
+import ac.uk.ebi.biostd.itest.itest.ITestListener.Companion.nfsSubmissionPath
+import ac.uk.ebi.biostd.itest.itest.ITestListener.Companion.tempFolder
+import ac.uk.ebi.biostd.itest.itest.getWebClient
+import ac.uk.ebi.biostd.persistence.common.service.SubmissionPersistenceQueryService
+import ac.uk.ebi.biostd.persistence.model.DbSequence
+import ac.uk.ebi.biostd.persistence.repositories.SequenceDataRepository
+import ebi.ac.uk.asserts.assertThat
+import ebi.ac.uk.dsl.tsv.line
+import ebi.ac.uk.dsl.tsv.tsv
+import ebi.ac.uk.extended.model.ExtAttribute
+import ebi.ac.uk.extended.model.ExtFile
+import ebi.ac.uk.extended.model.FireFile
+import ebi.ac.uk.extended.model.NfsFile
+import ebi.ac.uk.extended.model.StorageMode.FIRE
+import ebi.ac.uk.extended.model.StorageMode.NFS
+import ebi.ac.uk.io.ext.asFileList
+import ebi.ac.uk.io.ext.createFile
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.BeforeAll
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.context.SpringBootTest
+import org.springframework.boot.test.web.server.LocalServerPort
+import org.springframework.context.annotation.Import
+import org.springframework.test.context.junit.jupiter.SpringExtension
+
+@Import(PersistenceConfig::class)
+@ExtendWith(SpringExtension::class)
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+class SubmissionStorageModeTest(
+    @Autowired val securityTestService: SecurityTestService,
+    @Autowired val submissionRepository: SubmissionPersistenceQueryService,
+    @Autowired val sequenceRepository: SequenceDataRepository,
+    @LocalServerPort val serverPort: Int,
+) {
+    private lateinit var webClient: BioWebClient
+
+    @BeforeAll
+    fun init() {
+        securityTestService.ensureUserRegistration(SuperUser)
+        webClient = getWebClient(serverPort, SuperUser)
+
+        sequenceRepository.save(DbSequence("S-STR-MODE"))
+    }
+
+    @Test
+    fun `Fire to Nfs`() {
+        val submission = tsv {
+            line("Submission", "S-STR-MODE-1")
+            line("Title", "Storage mode submission")
+            line()
+
+            line("Study")
+            line("Type", "Experiment")
```

Comment:
I think it's important these two tests include File Lists since it's an important use case and, more importantly, they use a different mechanism for processing the file list itself.